### ISSUE TITLE
Run workflow After all package-level workflows have completed

### ISSUE THREAD

**Issue by TheSharpieOne** [circletron/circletron#8]
Currently, each package has it's own workflow. From what I can tell, this is how circletron should be configured. The problem is, workflows cannot depend on one-another, such that you cannot run something after all of the workflows from been completed (after all of the package stuff has been completed).
That said, I would like to run something after all of the packages have been completed. Right now, each package is responsible for versioning and creating change logs, which it commits and tags with the version. When that happens, we end up with lots of extra commits (one for each package changed).
![image](https://user-images.githubusercontent.com/664714/154133086-0667cb7e-e9bb-48f1-b4b6-b0ec4e4d87da.png)
Ideally, after all of the packages have done what they need to do, a single job will create the "release" commit with the change log, add all of the needed tags (for each of the new package version) and push that single commit up.
Not sure how to do that. One thought would be to have a single workflow defined in the root-level `circle.yml` which just references jobs in the package-level `circle.yml` (so that those jobs may be turned into `SKIP_JOB`s if needed. Each of those jobs would have the steps of what would be the workflow for that package. Then, have a job that requires those other jobs to be completed before it runs (this would be the "after all" job.) The downside is that root-level `circle.yml` would have to change if a new package is added (it has to know about all of the packages itself). Also, not sure if it would even work.

I was wondering if circletron could somehow (not sure how, but it might involve polling the API...) trigger a new workflow after the package-level workflows have completed.

Thoughts?

**Comment by insidewhy** [comment]
> The problem is, workflows cannot depend on one-another, such that you cannot run something after all of the workflows from been completed

I think this is a general problem with circleci... I'd really like it if circleci would allow the `requires` field of a job to refer to a job from a different workflow.

The only way to do this in circleci is to place all jobs in the workflow. So perhaps the solution that circletron should take would be to allow multiple `circle.yml` files to contribute to a single workflow.

So in this part of the code https://github.com/circletron/circletron/blob/main/src/index.ts#L185 instead of using `mergeObject` (which currently throws an error when two packages define the same workflow), it should use some kind of `mergeArraysIntoObject` which would concatenate the arrays instead of throwing.

Does that sound reasonable?

**Comment by TheSharpieOne** [comment]
Yeah, that sounds like it would solve the issue by creating a single workflow which allows us to `require` various package "steps" to complete before running additional jobs.
